webrtc-c/canary/src/CanaryWebrtc.cpp

```diff
@@ -91,11 +105,18 @@ STATUS run(Canary::PConfig pConfig)
         CHK_STATUS(peer.init(pConfig, callbacks));
         CHK_STATUS(peer.connect());
 
-        std::thread videoThread(sendLocalFrames, &peer, MEDIA_STREAM_TRACK_KIND_VIDEO, "./assets/h264SampleFrames/frame-%04d.h264",
-                                NUMBER_OF_H264_FRAME_FILES, SAMPLE_VIDEO_FRAME_DURATION);
-        std::thread audioThread(sendLocalFrames, &peer, MEDIA_STREAM_TRACK_KIND_AUDIO, "./assets/opusSampleFrames/sample-%03d.opus",
-                                NUMBER_OF_OPUS_FRAME_FILES, SAMPLE_AUDIO_FRAME_DURATION);
-
+        std::thread videoThread(sendLocalFrames, &peer, MEDIA_STREAM_TRACK_KIND_VIDEO, CANARY_VIDEO_FRAMES_PATH, NUMBER_OF_H264_FRAME_FILES,
+                                SAMPLE_VIDEO_FRAME_DURATION);
+        std::thread audioThread(sendLocalFrames, &peer, MEDIA_STREAM_TRACK_KIND_AUDIO, CANARY_AUDIO_FRAMES_PATH, NUMBER_OF_OPUS_FRAME_FILES,
+                                SAMPLE_AUDIO_FRAME_DURATION);
+
+        // All metrics tracking will happen on a time queue to simplify handling periodicity
+        CHK_STATUS(timerQueueAddTimer(timerQueueHandle, METRICS_INVOCATION_PERIOD, METRICS_INVOCATION_PERIOD, canaryRtpOutboundStats, (UINT64) &peer,
```

Comment:
For simplicity, I feel like we should combine all of these 3 callbacks into a single callback. `END_TO_END_METRICS_INVOCATION_PERIOD` can be equal to `METRICS_INVOCATION_PERIOD`.

Is there any reason why they're separated?

Reply:
The reason is if at any point of time, I want to change the period I extract each metrics, it would be easier if these were separate tasks. It is unlikely at this point, but just wanted to cover the case. We can always compress this if we need to later.

Reply:
I see That's fair